typing/typecore.ml

```diff
@@ -4770,247 +4773,335 @@ let spellcheck_idents ppf unbound valid_idents =
    To avoid confusion, it is disabled on float literals
    and when the expected type is `int` *)
-let report_literal_type_constraint h const =
-  let hint_const typ str_val =
+let report_literal_type_constraint ppf expected_type const =
+  let hint str_val =
     let hint_suffix c =
-      Some (fun ppf ->
-        fprintf ppf "@\n@[Hint: Did you mean `%s%c'?@]" str_val c)
+      fprintf ppf "@\n@[Hint: Did you mean `%s%c'?@]" str_val c
     in
-    if Path.same typ Predef.path_int32 then
+    if Path.same expected_type Predef.path_int32 then
       hint_suffix 'l'
-    else if Path.same typ Predef.path_int64 then
+    else if Path.same expected_type Predef.path_int64 then
       hint_suffix 'L'
-    else if Path.same typ Predef.path_nativeint then
+    else if Path.same expected_type Predef.path_nativeint then
       hint_suffix 'n'
-    else if Path.same typ Predef.path_float then
+    else if Path.same expected_type Predef.path_float then
       hint_suffix '.'
     else
-      None
-  in
-  let hint str_val =
-    match h with
-    | Ctype.Unification_trace.(Diff
-      { expected = { t = { desc = Tconstr (typ, [], _); _ }; _ }; _ }) ->
-        hint_const typ str_val
-    | _ -> None
+      ()
   in
   match const with
   | Const_int n -> hint (Int.to_string n)
   | Const_int32 n -> hint (Int32.to_string n)
   | Const_int64 n -> hint (Int64.to_string n)
   | Const_nativeint n -> hint (Nativeint.to_string n)
-  | _ -> None
+  | _ -> ()
 
-let report_literal_type_constraint ppf const trace =
-  let hints ~prev:_ h = report_literal_type_constraint h const in
-  begin match Ctype.Unification_trace.explain trace hints with
-  | Some hint -> hint ppf
-  | None -> ()
-  end
+let report_literal_type_constraint ppf const = function
+  | Some Unification_trace.
+    { expected = { t = { desc = Tconstr (typ, [], _) } } } ->
+      report_literal_type_constraint ppf typ const
+  | Some _ | None -> ()
 
-let report_expr_type_clash_hints ppf exp trace =
+let report_expr_type_clash_hints ppf exp diff =
   match exp with
-  | Some (Texp_constant const) -> report_literal_type_constraint ppf const trace
+  | Some (Texp_constant const) -> report_literal_type_constraint ppf const diff
   | _ -> ()
 
-let report_pattern_type_clash_hints ppf pat trace =
+let report_pattern_type_clash_hints ppf pat diff =
   match pat with
-  | Some (Tpat_constant const) -> report_literal_type_constraint ppf const trace
+  | Some (Tpat_constant const) -> report_literal_type_constraint ppf const diff
   | _ -> ()
 
+(* Hint when using int's operators (eg. `+`)
+   on other kind of integer and floats *)
+let report_numeric_operator_clash_hints ~loc env actual_type operator =
+  let stdlib = Path.Pident (Ident.create_persistent "Stdlib") in
+  let stdlib_p mod_ val_ = Path.Pdot (Path.Pdot (stdlib, mod_), val_) in
+  let hint expected_op =
+    Some (fun ppf ->
+      let expected_op = Env.normalize_path_prefix (Some loc) env expected_op in
```

Comment:
I get better printing if I remove this line (and ignore `env` and `loc`).

Reply:
Fixed, thanks !